mono/metadata/marshal.c

```diff
@@ -6518,7 +6603,34 @@ gboolean
 mono_method_has_unmanaged_callers_only_attribute (MonoMethod *method)
 {
 #ifndef ENABLE_NETCORE
-	return FALSE;
+	ERROR_DECL (attr_error);
+	MonoClass* attr_klass = NULL;
+	/* attribute is only allowed on static methods */
+	if (!m_method_is_static(method))
```

Comment:
Should we keep these early outs to speed up this check? Or should we let everything through and allow later "error" checks to handle these cases?

Reply:
I'm fine with keeping the early outs.

Reply:
Roslyn will error on these cases too, so only users doing bad custom IL should hit these failure cases, so as long as we do something other than randomly crash in hard to debug ways we should be fine.  What is the failure mode in those cases?

Reply:
Ideally, I would not have early outs. I'm just a bit worried about the cost of the attribute checks. Let me evaluate any impact again, and try to just leave a regular code path.

Reply:
I have removed the early outs, this means we will hit error code paths at other places in mono code. For example, when the ldftn is encountered it validates proper use of the attribute.

If we have specific error cases we care about, we should write tests in IL2CPP or Unity, and ensure all runtimes properly handle them.